src/address-manager/index.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+const debug = require('debug')
+const log = debug('libp2p:addresses')
+log.error = debug('libp2p:addresses:error')
+
+const MultiaddrSet = require('./multiaddr-set')
+
+/**
+ * Responsible for managing known peers, as well as their addresses, protocols and metadata.
+ */
+class AddressManager {
+  /**
+   * @constructor
+   * @param {object} [options]
+   * @param {Array<string>} [options.listen = []] list of multiaddrs string representation to listen.
+   * @param {Array<string>} [options.announce = []] list of multiaddrs string representation to announce.
+   * @param {Array<string>} [options.noAnnounce = []] list of multiaddrs string representation to not announce.
+   */
+  constructor ({ listen = [], announce = [], noAnnounce = [] } = {}) {
+    this._listen = new MultiaddrSet(listen)
```

Comment:
We could potentially just simplify this and store everything as strings internally, and return `Multiaddr`s whenever the data is requested. Then this could just be a `Set`.

Reply:
Totally agree, I went with this at first because I was already thinking ahead on when we will want to allow users to modify them. But no need to go with this for nothing now!